### PR TITLE
ARROW-7966: [FlightRPC][C++] Validate individual batches in integration

### PR DESCRIPTION
**Pull request by lidavidm**

### CODE REVIEW

File: cpp/src/arrow/flight/test_integration_client.cc [hunk 1]
```diff
@@ -106,17 +79,44 @@ arrow::Status UploadReaderToFlight(arrow::RecordBatchReader* reader,
 }
 
 /// \brief Helper to read a flight into a Table.
```
Comment (pitrou) [inline cpp/src/arrow/flight/test_integration_client.cc]: Update this docstring?

File: cpp/src/arrow/flight/test_integration_client.cc [hunk 2]
```diff
@@ -106,17 +79,44 @@ arrow::Status UploadReaderToFlight(arrow::RecordBatchReader* reader,
...
+    const std::vector<std::shared_ptr<arrow::RecordBatch>>& retrieved_data) {
   std::unique_ptr<arrow::flight::FlightClient> read_client;
   RETURN_NOT_OK(arrow::flight::FlightClient::Connect(location, &read_client));
 
   std::unique_ptr<arrow::flight::FlightStreamReader> stream;
   RETURN_NOT_OK(read_client->DoGet(ticket, &stream));
 
-  return ReadToTable(*stream, retrieved_data);
+  int counter = 0;
+  const int expected = retrieved_data.size();
```
Comment (pitrou) [inline cpp/src/arrow/flight/test_integration_client.cc]: Should probably `static_cast<int>` here.

**Review by pitrou — CHANGES_REQUESTED** [review]
Since you are doing this, perhaps `ValidateFull` should be called on each received record batch, both on the client and server side?

**Comment by lidavidm** [comment]
Thanks for the review Antoine! I've made the requested changes.

**Review by pitrou — APPROVED** [review]

**Comment by lidavidm** [comment]
There's a race in the test that triggered on Windows, I'll file a JIRA.

**Comment by pitrou** [comment]
> There's a race in the test that triggered on Windows, I'll file a JIRA.

Great. That failure appears occasionally on AppVeyor.

**Pull request closed without merging**